src/components/TooltipDropdown/TooltipDropdown.js

```diff
@@ -14,23 +14,29 @@ limitations under the License.
 import React from 'react';
 import { Dropdown, DropdownSkeleton } from 'carbon-components-react';
 
-const itemToElement = ({ text }) => {
+const itemToElement = ({ id, text }) => {
   return (
-    <div key={text} title={text}>
+    <div key={id} title={text}>
       {text}
     </div>
   );
 };
 
 const itemToString = ({ text }) => text;
 
-const itemStringToObject = text => ({ text });
+const itemToObject = item => {
```

Comment:
Would making all keys objects be better or overkill ?

Reply:
`item` can be either a string or an object. If it's an object we just return it as-is, if it's a string we transform it to be an object with the expected shape.

For most of our current use cases using strings is simpler, as the display value and the 'id' are the same. This change gives us flexibility where we want to override the 'id' (such as with the '*' for all namespaces).